reference-implementation/lib/transform-stream.js

```diff
@@ -79,21 +81,19 @@ function TransformStreamCloseReadableInternal(transformStream) {
   transformStream._readableClosed = true;
 }
 
-function TransformStreamResolveWrite(transformStream) {
-  if (transformStream._errored === true) {
-    return;
-  }
-
-  assert(transformStream._transforming === true);
-
-  assert(transformStream._resolveWrite !== undefined);
-
-  transformStream._transforming = false;
+function TransformStreamReadyPromise(transformStream) {
+  assert(transformStream._readyPromise_resolve === undefined);
 
-  transformStream._resolveWrite(undefined);
-  transformStream._resolveWrite = undefined;
+  let readyPromise;
```

Comment:
I think this would be easier to understand without the `readyPromise` variable. How about

```javascript
if (transformStream._readableBackpressure === false) {
    return Promise.resolve();
}
return new Promise(resolve => {
  transformStream._readyPromise_resolve = resolve;
});
```